cli/azd/cmd/auth_login.go

```diff
@@ -485,7 +503,7 @@ func (la *loginAction) login(ctx context.Context) error {
 					TenantID:     la.flags.tenantID,
 					RedirectPort: la.flags.redirectPort,
 					WithOpenUrl: func(url string) error {
-						openWithDefaultBrowser(ctx, la.console, url)
+						openWithDefaultBrowser(ctx, la.console, url, manualBrowsing)
```

Comment:
openWithDefaultBrowser is used from more than one place:
- auth Login
- monitor (open the portal links)
- help command (open the docs url)

When I created it, I was expecting to be a function which does not return an error and owns the entire responsibility of how azd handles opening urls, to keep it consistent.
I didn't want to handle the error on each part of the code where azd is trying to open a url, as the code would be the same.

The implementation of `openWithDefaultBrowser()` is already printing the url for the user when there's some error trying to launch the browser.
It also handles other special cases for browsing, like when BROSWER env var is set.

The other reason why I didn't want to return an error is because I want the function as such to be known as a function which can't really fail. Failing to launch a browser is not actually an error to handle for this function, but just an additional behavior which it knows what to do.

Reply:
To me the extra parameter seems overkill and overly verbose.  If we are able to detect that we cannot open a browser and we are in an interactive user session we should just print out the url and instructions for the user to open the browser to complete the login/auth flow.

Reply:
>  If we are able to detect that we cannot open a browser and ...

That's not the case. The use-case is:  We are able to open a browser, but the customer doesn't want to automatically open the browser.

When running on the Jupyter notebook, azd opens a new tab correctly when running `azd auth login --use-device-code`, but the feedback is that it is happening so fast, that the user didn't have time to copy the device code... and might not even know that there is a device-code.
So, the request is to print the url as an instruction instead of automatically opening the url.

But this is only for a special flow on Jupyper... we don't have feedback around this being an issue when running from the console or codespaces, because, in those cases, we ask customers to press enter, after copying the device-code, to open the login url. But for Jupyter, the `Enter` is not an option, so they need to use `--no-prompt` to make azd to skip that.

Reply:
I'm thinking, that, perhaps, I can re-use the flag `--no-prompt` as a sentinel here to activate the `--no-browser` mode.. as those 2 flags would basically go always together..  It would probably make no sense to see `azd auth login --use-device-code --no-browser`, as azd would ask for pressing enter, and then just giving an instruction after that for where to go